elfo-logger/src/actor.rs

```diff
@@ -120,61 +124,79 @@ impl Logger {
-            .and_then(|span_id| self.shared.spans.get(span_id))
-        {
-            span_id.clone_from(&data.parent_id);
+        let try_slow_path = {
+            let mut line = F::create_line(&mut self.buffer);
 
             let payload = self
                 .shared
                 .pool
-                .get(data.payload_id)
+                .get(event.payload_id)
                 .expect("unknown string");
 
-            T::Payload::fmt(out, &payload);
-        }
+            // <timestamp> <level> [<trace_id>] <object> - <message>\t<fields>
+
+            T::Timestamp::fmt(line.meta_mut(), &event.timestamp);
+            line.meta_mut().push(' ');
+            T::Level::fmt(line.meta_mut(), event.metadata.level());
+            line.meta_mut().push_str(" [");
+            T::TraceId::fmt(line.meta_mut(), &event.trace_id);
+            line.meta_mut().push_str("] ");
+            T::ActorMeta::fmt(line.payload_mut(), &event.object);
+            line.payload_mut().push_str(" - ");
+            T::Payload::fmt(line.payload_mut(), &payload);
+
+            // Add ancestors' fields.
+            let mut span_id = event.span_id.clone();
+
+            {
+                let payload_buffer = line.payload_mut();
+                while let Some(data) = span_id
+                    .as_ref()
+                    .and_then(|span_id| self.shared.spans.get(span_id))
+                {
+                    span_id.clone_from(&data.parent_id);
+
+                    let payload = self
+                        .shared
+                        .pool
+                        .get(data.payload_id)
+                        .expect("unknown string");
+
+                    T::Payload::fmt(payload_buffer, &payload);
+                }
+            }
 
-        if config.format.with_location {
-            if let Some(location) = extract_location(event.metadata) {
-                out.push('\t');
-                T::Location::fmt(out, &location);
+            if config.format.with_location {
+                if let Some(location) = extract_location(event.metadata) {
+                    let fields_buffer = line.fields_mut();
+                    fields_buffer.push('\t');
+                    T::Location::fmt(line.fields_mut(), &location);
+                }
             }
-        }
 
-        if config.format.with_module {
-            if let Some(module) = event.metadata.module_path() {
-                out.push('\t');
-                T::Module::fmt(out, module);
+            if config.format.with_module {
+                if let Some(module) = event.metadata.module_path() {
+                    let fields_buffer = line.fields_mut();
+                    fields_buffer.push('\t');
+                    T::Module::fmt(fields_buffer, module);
+                }
             }
-        }
 
-        out.push('\n');
+            let try_slow_path =
+                !F::STOP && (line.total_wrote() > self.ctx.config().max_line_size.0 as _);
```

Comment:
It looks error-prone to have logic based on `max_line_size` in two different places: here and in `LineBuffer`.